### ISSUE TITLE
oh-my-zsh completions

### ISSUE THREAD

**Issue by sdykae** [typst/typst#4333]
### Description

I have this completion for oh my zsh, somehow the current install with brew for mac does not have directory completions for the compile command or others, did not found any fixes or how to do it, bash works well but oh-my-zsh does not, zsh raw also works, so I made this completion script to place in the completions dir in the oh-my-zsh config dir.

It just works, 
maybe someone experienced with the repo could make this a feature but now the completion works with it

`/Users/myuser/.oh-my-zsh/completions/_typst`
```bash
#compdef typst

_typst() {
  local curcontext="$curcontext" state line
  typeset -A opt_args

  _arguments -C \
    '1: :->command' \
    '*:: :->args'

  case $state in
    command)
      _values 'typst command' \
        'compile[c]' \
        'fonts' \
        'help' \
        'init' \
        'query' \
        'update' \
        'watch[w]'
      ;;
    args)
      case $line[1] in
        compile|c)
          _files
          ;;
        *)
          _message 'No more arguments'
          ;;
      esac
      ;;
  esac
}

_typst "$@"

```

### Use Case

Path, directory, files completion for oh-my-zsh

**Comment by Enivex** [comment]
Is this a duplicate of https://github.com/typst/typst/issues/4156 ?

**Comment by sdykae** [comment]
> Is this a duplicate of https://github.com/typst/typst/issues/4156 ?

That one is for `nix` only :,)

**Comment by alerque** [comment]
Yes this is a duplicate of #4156. It is not specific to oh-my-zsh, the ZSH completions are just completely broken across the board. The other issue mentioned testing with Nix, but the problem is not specific to Nix (I duplicated this on Arch Linux with the last stable release package there).
